Run real networking tests via sanitycheck

pfalcon opened the pull request:
This patch combines:

* network loopback interface, #4461
* support for running "network" QEMU without connecting to host for networking (instead using just the internal loopback iface), https://github.com/zephyrproject-rtos/zephyr/pull/4457
* suitable testcase.yaml settings (thanks to @nashif for hints)

- to allow to run "real", "full IP stack" networking tests via sanitycheck and thus standard CI systems.

nashif commented on tests/net/socket/udp/testcase.yaml:
```diff
@@ -1,5 +1,5 @@
 tests:
   - test:
-      build_only: true
+      extra_args: CONF_FILE="prj.conf prj-sanitycheck.conf"
```
you can avoid this with:

```
extra_configs:
 - CONFIG_NET_TEST=y
 - CONFIG_NET_LOOPBACK=y

```

Then you will not need the `prj-sanitycheck.conf` file

nashif replied:
this replaces extra_args here

pfalcon replied:
Nice hack, will update.

(Testing how it actually runs in CI so far).

pfalcon commented:
This is pending on https://github.com/zephyrproject-rtos/zephyr/pull/4457 and https://github.com/zephyrproject-rtos/zephyr/pull/4621 to be fully ready.

pfalcon commented:
Ok, now depends only on #4457 and should be good to review/merge.

nashif commented on scripts/Makefile.qemu:
```diff
@@ -26,4 +26,8 @@ run: zephyr
+	$(Q)if [ "$(CONFIG_NET_TEST)" = "y" ]; then \
+	$(QEMU) $(QEMU_FLAGS) -kernel $(KERNEL_ELF_NAME); \
+	else \
+	$(QEMU) $(QEMU_FLAGS) $(QEMU_EXTRA_FLAGS) -kernel $(KERNEL_ELF_NAME); \
```
who sets QEMU_EXTRA_FLAGS? Why dont you unset it in there instead of changing this here in a top level Makefile that has nothing to do with networking?
Maybe in samples/net/common/Makefile.ipstack?

pfalcon committed:
tests: socket: Make runnable with sanitycheck

Relies on loopback network interface added recently and being able
to run QEMU without connecting to host networking.

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

pfalcon commented:
This patch has been updated to the latest master. With CMake migration, needed infrastructure "just works". So, this patch only updates testcase.yaml's.

@nashif , @jukkar : Please re-review.
